client/my-sites/plans-features-main/index.jsx

```diff
@@ -602,16 +555,15 @@ export class PlansFeaturesMain extends Component {
 		return props.planTypeSelector;
 	}
 
-	renderPlansGrid() {
+	renderPlansGrid( plans, visiblePlans ) {
 		const { shouldShowPlansFeatureComparison, is2023PricingGridVisible } = this.props;
 
-		if ( is2023PricingGridVisible ) {
-			return this.show2023OnboardingPricingGrid();
+		// TODO: need to figure out if we can deprecate `shouldShowPlansFeatureComparison`
+		if ( is2023PricingGridVisible || shouldShowPlansFeatureComparison ) {
```

Comment:
This conditional means that the new pricing grid could possibly be shown in cases where `is2023PricingGridVisible` is false and `shouldShowPlansFeatureComparison` is true. This is an undocumented "feature", and I think we should solely rely on `is2023PricingGridVisible` to decide the visibility of the new grid since that's well documented.

If this case has actually happened (`is2023PricingGridVisible` = false and `shouldShowPlansFeatureComparison` = true), and the new pricing grid is being shown, then that should be considered a bug since it's unintentional.

I think we should remove this var from the conditional, and in the future, get rid of it altogether. 🙂

Reply:
Great catch! It's actually a quite major logical flaw. 

After digging a bit more and investigating [how plans-features-main-pm uses this prorperty](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/plans-pm/plans-features-main-pm.jsx#L45), which happens to be a copy of how things used to be, I now understand that that prop is used to mainly determine whether to render `PlansFeaturesComparison` or `PlansFeatures`; the former is for the desktop, and the latter is for the mobile.

To better clarify it, I've done some more rearragnement in https://github.com/Automattic/wp-calypso/pull/76070/commits/3cf40524a692a095997880d04cdb2cb0136ae29d to make this clearer by introducing a `renderLegacyPricingGrid()` function. 

I've also added a note that the legacy component should be loaded in async as a follow-up, which might have quite some implication on the page load speed.